File: editor/en/[442]Find All Duplicates in an Array.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        result = []
        isDupicated = [False] * (len(nums) + 1)

        for num in nums:
            if isDupicated[num]:
                result.append(num)
            else:
                isDupicated[num] = True

        return result

# leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[442]Find All Duplicates in an Array.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> result = new ArrayList<>();
        boolean[] isDuplicated = new boolean[nums.length + 1];

        for (int num : nums) {
            if (isDuplicated[num]) {
                result.add(num);
            } else {
                isDuplicated[num] = true;
            }
        }

        return result;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
